main.rkt

```diff
@@ -91,6 +91,8 @@
   (define *namespace* (make-parameter "main"))
   (define *in-file* (make-parameter "-"))
   (define *out-file* (make-parameter "-"))
+  (define *filter-in* (make-parameter "-"))
+  (define *filter-out* (make-parameter "-"))
   (define *rel-error* (make-parameter #f))
```

Comment:
Why do these have to be parameters? Can't they just be local variables in `infra/filter.rkt`?

Reply:
I believe you are correct...fixing.

Reply:
Actually I think we might need it because it follows the same pattern as evaluate.

Reply:
Why not just use in-file out-file?

Reply:
Do we make it mandatory they are passed in then?

Reply:
Also, Just tried using in_file and out_file but it didn't compile because I believe those can only be used in the scope of those subarguments. Gonna leave as is.